### ISSUE TITLE
[Windows] test_distutils leaks a vc140.pdb file

### ISSUE THREAD

**Issue by vstinner** [python/cpython#74318]
BPO | [30132](https://bugs.python.org/issue30132)
--- | :---
Nosy | @pfmoore, @vstinner, @tjguk, @zware, @zooba
PRs | <li>python/cpython#1278</li><li>python/cpython#1380</li><li>python/cpython#1387</li><li>python/cpython#1388</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-05-02.15:26:45.584>
created_at = <Date 2017-04-21.16:17:40.329>
labels = ['3.7', 'tests', 'OS-windows', 'performance']
title = '[Windows] test_distutils leaks a vc140.pdb file'
updated_at = <Date 2017-05-02.15:26:45.582>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-05-02.15:26:45.582>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-05-02.15:26:45.584>
closer = 'vstinner'
components = ['Tests', 'Windows']
creation = <Date 2017-04-21.16:17:40.329>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 30132
keywords = []
message_count = 9.0
messages = ['292060', '292244', '292247', '292255', '292740', '292745', '292756', '292763', '292780']
nosy_count = 5.0
nosy_names = ['paul.moore', 'vstinner', 'tim.golden', 'zach.ware', 'steve.dower']
pr_nums = ['1278', '1380', '1387', '1388']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'resource usage'
url = 'https://bugs.python.org/issue30132'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
http://buildbot.python.org/all/builders/AMD64%20Windows8%203.x/builds/566/steps/test/logs/stdio

Warning -- files was modified by test_distutils
  Before: []
  After:  ['vc140.pdb']

**Comment by vstinner** [comment]
Ok, I found the guilty: test_build_ext, see linked PR which should fix the warning.

**Comment by vstinner** [comment]
New changeset f6448e5d65c349576df6e83b8324b9c208e77615 by Victor Stinner in branch 'master':
bpo-30132: distutils test_build_ext() uses temp_cwd() (bpo-1278)
https://github.com/python/cpython/commit/f6448e5d65c349576df6e83b8324b9c208e77615

**Comment by vstinner** [comment]
Reopen, while the warning was fixed on AppVeyor and my Windows VM, I still see the warning on Windows buildbots.

http://buildbot.python.org/all/builders/AMD64%20Windows8%203.x/builds/578/steps/test/logs/stdio

Warning -- files was modified by test_distutils
  Before: []
  After:  ['vc140.pdb']

http://buildbot.python.org/all/builders/AMD64%20Windows10%203.x/builds/591/steps/test/logs/stdio

Warning -- files was modified by test_distutils
  Before: []
  After:  ['vc140.pdb']

**Comment by vstinner** [comment]
I'm unable to run test_distutils on Windows using python_d.exe: I get link error 1101.

LINK : fatal error LNK1101: incorrect MSPDB140.DLL version; recheck installation of this product

I ran manually the following command, but it didn't help:

   "%VS140COMNTOOLS%\\..\\..\\VC"\\vcvarsall.bat amd64 

I also tried to modify manually the PATH environment variable to only include the VC amd64 path, but it didn't work neither. I don't know how to fix the error. I don't understand how the configure the C compiler and it's hard to get good documentation on it :-/

So I wrote https://github.com/python/cpython/pull/1380 another attempt to fix the test_distutils warning.

**Comment by vstinner** [comment]
New changeset 92fd6c9ef54a857815e3dca8ee74b2b4f5cdf154 by Victor Stinner in branch 'master':
bpo-30132: distutils BuildExtTestCase use temp_cwd (bpo-1380)
https://github.com/python/cpython/commit/92fd6c9ef54a857815e3dca8ee74b2b4f5cdf154

**Comment by vstinner** [comment]
New changeset 30768958490c658fba0fe24f1cabbdad44be22ff by Victor Stinner in branch '3.6':
bpo-30132: distutils BuildExtTestCase use temp_cwd (bpo-1387)
https://github.com/python/cpython/commit/30768958490c658fba0fe24f1cabbdad44be22ff

**Comment by vstinner** [comment]
New changeset 3ceeb84b5a90e327ef55190105e82c926b0b8aab by Victor Stinner in branch '3.5':
bpo-30132: distutils BuildExtTestCase use temp_cwd (bpo-1387) (bpo-1388)
https://github.com/python/cpython/commit/3ceeb84b5a90e327ef55190105e82c926b0b8aab

**Comment by vstinner** [comment]
Ok, this issue is now fixed in 3.5, 3.6 and master (3.7).
